src/viewer/SatelliteGroup.ts

```diff
@@ -83,10 +83,12 @@ class SatelliteGroup {
    * @throws {Error} If nameRegex is not a string.
    */
   private searchNameRegex () {
-    if (typeof this.data !== 'string') {
-      throw new Error('nameRegex must be a string');
+    if (typeof this.data === 'string') {
+      throw new Error('nameRegex must be a RegExp');
```

Comment:
Actually there may be a small issue, since we are now loading the groups from the `config.json`, since here the regular expressions are strings and we haven't done any conversions to a RegExp. For this reason we should probably support both strings and RegExp, converting any string to RegExp:

```ts
    let regex: RexExp; 
    if (typeof this.data === 'string') {
      regex = new RegExp(this.data);
    } else if (this.data instanceof RegExp) {
      regex = this.data;
    } else {
      throw new Error('nameRegex must be a RegExp or a regex in string form');
    }
```